swarm/network/stream/syncer_test.go

```diff
@@ -65,203 +68,178 @@ func testSyncBetweenNodes(t *testing.T, nodes, conns, chunkCount int, skipCheck
 	defer setDefaultSkipCheck(defaultSkipCheck)
 	defaultSkipCheck = skipCheck
 	//data directories for each node and store
-	datadirs = make(map[discover.NodeID]string)
-	if *useMockStore {
-		createStoreFunc = createMockStore
-		createGlobalStore()
-	} else {
-		createStoreFunc = createTestLocalStorageFromSim
-	}
-	defer datadirsCleanup()
+	//datadirs = make(map[discover.NodeID]string)
+	//TODO fix mockstore stuff
+	/*
+		if *useMockStore {
+			createStoreFunc = createMockStore
+			createGlobalStore()
+		} else {
+			createStoreFunc = createTestLocalStorageFromSim
+		}
+	*/
+
+	sim := simulation.New(map[string]simulation.ServiceFunc{
+		"streamer": func(ctx *adapters.ServiceContext, bucket *sync.Map) (s node.Service, cleanup func(), err error) {
+
+			id := ctx.Config.ID
+			addr := network.NewAddrFromNodeID(id)
+			//hack to put addresses in same space
+			addr.OAddr[0] = byte(0)
+
+			store, datadir, err := createTestLocalStorageForId(id, addr)
+			if err != nil {
+				return nil, nil, err
+			}
+			bucket.Store(bucketKeyStore, store)
+			cleanup = func() {
+				store.Close()
+				os.RemoveAll(datadir)
+			}
+			localStore := store.(*storage.LocalStore)
+			db := storage.NewDBAPI(localStore)
+			bucket.Store(bucketKeyDB, db)
+			kad := network.NewKademlia(addr.Over(), network.NewKadParams())
+			delivery := NewDelivery(kad, db)
+			bucket.Store(bucketKeyDelivery, delivery)
+
+			r := NewRegistry(addr, delivery, db, state.NewInmemoryStore(), &RegistryOptions{
+				SkipCheck:  defaultSkipCheck,
+				DoRetrieve: false,
+			})
+			RegisterSwarmSyncerServer(r, db)
```

Comment:
RegisterSwarmSyncerServer and RegisterSwarmSyncerClient are not required here. They are called in NewRegistry.

Reply:
Is this for all stream tests using the `streamer` test or only for `syncer_test.go`?

Reply:
It is even more general, NewRegistry is calling this two functions, so when a registry is created by NewRegistry, this two functions are not required.